src/wasm-builder.h

```diff
@@ -33,10 +33,11 @@ struct NameType {
 // General AST node builder
 
 class Builder {
+  Module& wasm;
   MixedArena &allocator;
 
 public:
-  Builder(Module& wasm) : allocator(wasm.allocator) {}
+  Builder(Module& wasm) : wasm(wasm), allocator(wasm.allocator) {}
```

Comment:
if we need to add the module here, which seems reasonable, then let's get rid of the allocator. we don't need both.